host/loadelf.c

```diff
@@ -144,6 +149,13 @@ static oe_result_t _oe_load_elf_image(
         {
             OE_RAISE(OE_FAILURE);
         }
+
+        /* It is now the default for linux shared libraries and executables to
+         * have the build-id note. GCC by default passes the --build-id option
+         * to linker, whereas clang does not. Build-id is also used as keys by
+         * debug symbol-servers.*/
+        if (!has_build_id)
```

Comment:
Should this be a fatal error? I can imagine that there are cases where the dev simply doesn't care if the enclave is debuggable. 

This is also a breaking change because it prevents old enclaves from loading with the new host, which should be called out in the change log.

Reply:
I am undecided about whether this should be a fatal error. 
Old enclaves built using GCC always had the build-id; but after the switch to clang, we lost the build-ids.

We can make it a non fatal check and have the enclave debuggers print out a message if the build-id note is not found.